js/flatbuffers.js

```diff
@@ -677,11 +677,11 @@ outer_loop:
  * @param {flatbuffers.Offset} root_table
  * @param {string=} opt_file_identifier
  */
-flatbuffers.Builder.prototype.finish = function(root_table, opt_file_identifier) {
+flatbuffers.Builder.prototype.finish = function(root_table, opt_file_identifier, size_prefix) {
   if (opt_file_identifier) {
     var file_identifier = opt_file_identifier;
     this.prep(this.minalign, flatbuffers.SIZEOF_INT +
-      flatbuffers.FILE_IDENTIFIER_LENGTH);
+      flatbuffers.FILE_IDENTIFIER_LENGTH + (size_prefix ? SIZEOF_INT : 0));
```

Comment:
This size gets added only if `opt_file_identifier` is true, but the corresponding data get written always?

Reply:
Mmm yes that can't be right, I must consider all overloads from https://github.com/robert-schmidtke/flatbuffers/blob/3aac3689b5bea99148b8b5d76644322e2a0148db/java/com/google/flatbuffers/FlatBufferBuilder.java#L821-L877.

Reply:
I think I got it right with https://github.com/google/flatbuffers/pull/5326/commits/ba77cc3eac427859d67515eba94674815b90e5dd.